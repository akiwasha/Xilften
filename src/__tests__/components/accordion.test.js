import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import faqsData from '../../fixtures/faqs.json';
import { Accordion } from '../../components';

describe('<Accordion />', () => {
  it('renders the <Accordion /> with the populated data', () => {
    const { container, getByText } = render(
      <Accordion>
        <Accordion.Title>Foire aux questions</Accordion.Title>
        {faqsData.map((item) => (
          <Accordion.Item key={item.id}>
            <Accordion.Header>{item.header}</Accordion.Header>
            <Accordion.Body>{item.body}</Accordion.Body>
          </Accordion.Item>
        ))}
      </Accordion>
    );

    expect(getByText('Foire aux questions')).toBeTruthy();
    expect(getByText("xilften, qu'est-ce que c'est ?")).toBeTruthy();
    expect(getByText('Combien coûte xilften ?')).toBeTruthy();
    expect(getByText('Où puis-je regarder xilften ?')).toBeTruthy();
    expect(getByText('Comment puis-je annuler mon forfait ?')).toBeTruthy();
    expect(getByText('Que puis-je regarder sur xilften ?')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('opens and closes the <Accordion /> component', () => {
    const { container, queryByText } = render(
      <Accordion>
        <Accordion.Title>Foire aux questions</Accordion.Title>
        {faqsData.map((item) => (
          <Accordion.Item key={item.id}>
            <Accordion.Header>{item.header}</Accordion.Header>
            <Accordion.Body>{item.body}</Accordion.Body>
          </Accordion.Item>
        ))}
      </Accordion>
    );

    const whatIsxilftenBodyText =
      "xilften est un service de streaming qui propose une vaste sélection de séries TV, films, animes, documentaires et autres programmes primés sur des milliers d'appareils connectés à Internet. Regardez tout ce que vous voulez, quand vous voulez, sans publicité et à un tarif mensuel très attractif. Découvrez de nouveaux films et séries TV chaque semaine, il y en a pour tous les goûts !";

    expect(queryByText(whatIsxilftenBodyText)).toBeFalsy();
    fireEvent.click(queryByText("xilften, qu'est-ce que c'est ?"));
    expect(queryByText(whatIsxilftenBodyText)).toBeTruthy();

    fireEvent.click(queryByText("xilften, qu'est-ce que c'est ?"));
    expect(queryByText(whatIsxilftenBodyText)).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
