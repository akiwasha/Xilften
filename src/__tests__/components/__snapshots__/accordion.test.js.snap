// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> opens and closes the <Accordion /> component 1`] = `
<div
  class="accordion__Container-qskqv3-0 bKPrvs"
>
  <div
    class="accordion__Inner-qskqv3-1 ilrgEX"
  >
    <h1
      class="accordion__Title-qskqv3-5 kVSpkJ"
    >
      Foire aux questions
    </h1>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        xilften, qu'est-ce que c'est ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        Combien co청te xilften ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        O첫 puis-je regarder xilften ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        Comment puis-je annuler mon forfait ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        Que puis-je regarder sur xilften ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Accordion /> renders the <Accordion /> with the populated data 1`] = `
<div
  class="accordion__Container-qskqv3-0 bKPrvs"
>
  <div
    class="accordion__Inner-qskqv3-1 ilrgEX"
  >
    <h1
      class="accordion__Title-qskqv3-5 kVSpkJ"
    >
      Foire aux questions
    </h1>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        xilften, qu'est-ce que c'est ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        Combien co청te xilften ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        O첫 puis-je regarder xilften ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        Comment puis-je annuler mon forfait ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
    <div
      class="accordion__Item-qskqv3-2 kIaGRB"
    >
      <div
        class="accordion__Header-qskqv3-3 jZvcjb"
      >
        Que puis-je regarder sur xilften ?
        <img
          alt="Open"
          src="/images/icons/add.png"
        />
      </div>
    </div>
  </div>
</div>
`;
